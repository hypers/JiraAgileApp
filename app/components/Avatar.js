import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import {
  View,
  Image,
  Platform
} from 'react-native';

import SvgUri from './SvgUri';

export default class SliderEntry extends PureComponent {
  render() {
    let avatar = null;
    const { uri, width, height, style } = this.props;
    if (uri) {
      if (uri.indexOf('ownerId') >= 0 ) {
        avatar = <Image
          style={[style, {
            width,
            height
          }]}
          source={{ uri }}
        /> ;
      } else {
        avatar = <SvgUri
          style={style}
          source={{ uri }}
          width={width}
          height={height}
        /> ;
      }
    } else {
      avatar = <SvgUri
        style={style}
        source={require('../resources/useravatar.svg')}
        width={width}
        height={height}
      /> ;
    }
    return avatar;
  }
}