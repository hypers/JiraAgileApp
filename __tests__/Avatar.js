import 'react-native';
import React from 'react';

import Avatar from '../app/components/Avatar';


// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

beforeAll(() => {
  require('isomorphic-fetch');
  global.jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;
});

describe.skip('Avatar', () => {

  // 自定义的 png 格式头像
  test('Avatar custom png', () => {
    const tree = renderer.create(
      <Avatar
        uri='https://jira.hypers.com/secure/useravatar?size=small&ownerId=jianbo.yu&avatarId=11718'
      />
    );
    expect(tree).toMatchSnapshot();
  });

  // svg 格式头像
  test('Avatar default svg', done => {
    const tree = renderer.create(
      <Avatar
        uri='https://jira.hypers.com/secure/useravatar?size=small&avatarId=10122'
      />
    );
    expect(tree).toMatchSnapshot();
    setTimeout(() => {
      expect(tree).toMatchSnapshot();
      done();
    }, 5000);
  });

  // svg 的 uri 错误
  test('Avatar svg uri error', done => {
    const tree = renderer.create(
      <Avatar
        uri='https://jiraa.hypersS.com/secure/useravatar?size=small&avatarId=4367467526'
      />
    );
    expect(tree).toMatchSnapshot();
    setTimeout(() => {
      expect(tree).toMatchSnapshot();
      done();
    }, 5000);
  });

  // 无 uri
  test('Avatar uri is undefined', done => {
    const tree = renderer.create(
      <Avatar />
    );
    expect(tree).toMatchSnapshot();
  });


});

