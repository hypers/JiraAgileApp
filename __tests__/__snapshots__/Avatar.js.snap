// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Avatar custom png 1`] = `
<Image
  source={
    Object {
      "uri": "https://jira.hypers.com/secure/useravatar?size=small&ownerId=jianbo.yu&avatarId=11718",
    }
  }
  style={
    Array [
      undefined,
      Object {
        "height": undefined,
        "width": undefined,
      },
    ]
  }
/>
`;

exports[`Avatar Avatar default svg 1`] = `null`;

exports[`Avatar Avatar default svg 2`] = `
<View
  style={undefined}
>
  <RNSVGSvgView
    align="xMidYMid"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        false,
        undefined,
      ]
    }
    vbHeight={24}
    vbWidth={24}
    x="0"
    y="0"
  >
    <RNSVGGroup
      fill={
        Array [
          0,
          0,
          0,
          0,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      name="color_x5F_bg"
      opacity={1}
      propList={Array []}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={null}
    >
      <RNSVGPath
        d="M12,24L12,24C5.4,24,0,18.6,0,12l0,0C0,5.4,5.4,0,12,0l0,0c6.6,0,12,5.4,12,12l0,0C24,18.6,18.6,24,12,24z"
        fill={
          Array [
            0,
            0.9254901960784314,
            0.9490196078431372,
            0.9764705882352941,
            1,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        propList={
          Array [
            "fill",
          ]
        }
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={null}
      />
    </RNSVGGroup>
    <RNSVGPath
      d="M17.7,6.5h-1c0-0.9-0.8-1.6-1.6-1.6H7.5C6.7,4.9,6,5.6,6,6.4v0.5C5.3,7,4.9,7.6,4.9,8.2v4.4H6v0.5h10.7V13
    	h2.6V8.2C19.3,7.2,18.6,6.5,17.7,6.5z"
      fill={
        Array [
          0,
          0.6745098039215687,
          0.7490196078431373,
          0.803921568627451,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      propList={
        Array [
          "fill",
        ]
      }
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={null}
    />
    <RNSVGPath
      d="M18.4,11.1L18.4,11.1c-0.9,0-1.6-0.8-1.6-1.6V8.6c0,0-0.1-0.2-0.1,0c-0.3,0.5-0.8,0.9-1.4,0.9H6.4
    	c-0.4,0-0.8,0.3-0.8,0.8v0.9c-0.9,0-1.6,0.8-1.6,1.6c0,0.9,0.7,1.6,1.6,1.6c0.1,3.3,2.6,6.1,5.9,6.3c3.7,0.2,6.8-2.7,6.8-6.3l0,0
    	c0.9,0,1.6-0.8,1.6-1.6C20,11.8,19.2,11.1,18.4,11.1z"
      fill={
        Array [
          0,
          0.7764705882352941,
          0.8313725490196079,
          0.8745098039215686,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      propList={
        Array [
          "fill",
        ]
      }
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={null}
    />
    <RNSVGGroup
      fill={
        Array [
          0,
          0,
          0,
          0,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      name="Layer_2"
      opacity={1}
      propList={Array []}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={null}
    />
  </RNSVGSvgView>
</View>
`;

exports[`Avatar Avatar svg uri error 1`] = `null`;

exports[`Avatar Avatar svg uri error 2`] = `null`;

exports[`Avatar Avatar uri is undefined 1`] = `
<View
  style={undefined}
>
  <RNSVGSvgView
    align="xMidYMid"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        false,
        undefined,
      ]
    }
    vbHeight={24}
    vbWidth={24}
    x="0"
    y="0"
  >
    <RNSVGGroup
      fill={
        Array [
          0,
          0,
          0,
          0,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      name="color_x5F_bg"
      opacity={1}
      propList={Array []}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={null}
    >
      <RNSVGPath
        d="M12,24L12,24C5.4,24,0,18.6,0,12l0,0C0,5.4,5.4,0,12,0l0,0c6.6,0,12,5.4,12,12l0,0C24,18.6,18.6,24,12,24z"
        fill={
          Array [
            0,
            0.9254901960784314,
            0.9490196078431372,
            0.9764705882352941,
            1,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        propList={
          Array [
            "fill",
          ]
        }
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={null}
      />
    </RNSVGGroup>
    <RNSVGPath
      d="M17.7,6.5h-1c0-0.9-0.8-1.6-1.6-1.6H7.5C6.7,4.9,6,5.6,6,6.4v0.5C5.3,7,4.9,7.6,4.9,8.2v4.4H6v0.5h10.7V13
    	h2.6V8.2C19.3,7.2,18.6,6.5,17.7,6.5z"
      fill={
        Array [
          0,
          0.6745098039215687,
          0.7490196078431373,
          0.803921568627451,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      propList={
        Array [
          "fill",
        ]
      }
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={null}
    />
    <RNSVGPath
      d="M18.4,11.1L18.4,11.1c-0.9,0-1.6-0.8-1.6-1.6V8.6c0,0-0.1-0.2-0.1,0c-0.3,0.5-0.8,0.9-1.4,0.9H6.4
    	c-0.4,0-0.8,0.3-0.8,0.8v0.9c-0.9,0-1.6,0.8-1.6,1.6c0,0.9,0.7,1.6,1.6,1.6c0.1,3.3,2.6,6.1,5.9,6.3c3.7,0.2,6.8-2.7,6.8-6.3l0,0
    	c0.9,0,1.6-0.8,1.6-1.6C20,11.8,19.2,11.1,18.4,11.1z"
      fill={
        Array [
          0,
          0.7764705882352941,
          0.8313725490196079,
          0.8745098039215686,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      propList={
        Array [
          "fill",
        ]
      }
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={null}
    />
    <RNSVGGroup
      fill={
        Array [
          0,
          0,
          0,
          0,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      name="Layer_2"
      opacity={1}
      propList={Array []}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={null}
    />
  </RNSVGSvgView>
</View>
`;
